strawberry/dataloader.py

```diff
@@ -0,0 +1,80 @@
+import dataclasses
+from asyncio import create_task, get_event_loop
+from asyncio.futures import Future
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, Generic, List, Optional, TypeVar
+
+
+T = TypeVar("T")
+K = TypeVar("K")
+
+
+@dataclass
+class LoaderTask:
+    key: Any
+    future: Future
+
+
+@dataclass
+class Batch:
+    tasks: List[LoaderTask] = dataclasses.field(default_factory=list)
+
+    def add_task(self, key: Any, future: Future):
+        task = LoaderTask(key, future)
+        self.tasks.append(task)
+
+
+class DataLoader(Generic[T, K]):
+    queue: List[LoaderTask] = []
+    batch: Optional[Batch] = None
+
+    def __init__(self, load_fn: Callable):
+        self.load_fn = load_fn
+
+        self.loop = get_event_loop()
+
+    def load(self, key: K) -> Awaitable:
+        future = self.loop.create_future()
+
+        batch = get_current_batch(self)
+        batch.add_task(key, future)
+
+        return future
+
+
+def should_create_new_batch(batch: Batch) -> bool:
+    return False
```

Comment:
This should return true if the batch is being scheduled or if we reached the max items per batch